routers/web/user/profile.go

```diff
@@ -90,6 +91,46 @@ func Profile(ctx *context.Context) {
 		ctx.Data["RenderedDescription"] = content
 	}
 
+	repo, err := repo_model.GetRepositoryByName(ctx.ContextUser.ID, ".profile")
+	if err == nil && !repo.IsEmpty {
+		gitRepo, err := git.OpenRepository(ctx, repo.RepoPath())
+		if err != nil {
+			ctx.ServerError("OpenRepository", err)
+		}
+		defer gitRepo.Close()
+		defaultBranch := repo.DefaultBranch // TODO: get default branch from repo module
+		commitID, err := gitRepo.GetBranchCommitID(defaultBranch)
+		if err != nil {
+			ctx.ServerError("GetBranchCommitID", err)
+		}
+		commit, err := gitRepo.GetCommit(commitID)
+		if err != nil {
+			ctx.ServerError("GetCommit", err)
+		}
+		tree, err := commit.SubTree("")
+
+		if err != nil || tree == nil {
+			ctx.ServerError("SubTree", err)
+		}
+		blob, err := tree.GetBlobByPath("README.md")
+		if err != nil {
+			ctx.ServerError("GetBlobByPath", err)
+		}
+		bytes, err := blob.GetBlobContent()
+		if err != nil {
+			ctx.ServerError("GetBlobContent", err)
+		}
+		profileContent, err := markdown.RenderString(&markup.RenderContext{
+			Ctx:     ctx,
+			GitRepo: gitRepo,
+		}, bytes)
+		if err != nil {
+			ctx.ServerError("RenderString", err)
+			return
```

Comment:
I think you should take a look at routers/web/repo/render.go:RenderFile.

But you probably should set a limit on the the size of the file that is rendered here.

Reply:
There is an inherent limit it seems when using markdown.RenderString, as to what the exact limit is, I would have to investigate.

Reply:
Also curious as to why using renger.go:RenderFile would be a better approach than the current method.

Reply:
@IDeletedSystem64 See above